src/poly/commitment.rs

```diff
@@ -279,14 +279,13 @@ fn test_opening_proof() {
     transcript.absorb(Fp::from_bytes(&v.to_bytes()).unwrap()); // unlikely to fail since p ~ q
 
     loop {
-        let transcript_dup = transcript.clone();
+        let mut transcript_dup = transcript.clone();
 
         let opening_proof = Proof::create(&params, &mut transcript, &px, blind, x);
         if opening_proof.is_err() {
```

Comment:
Style nit: use matching.

Reply:
Agreed and in fact I think this should just be changed to use matching.

Reply:
clippy is still complaining, and I agree.

Reply:
Agreed; this should be:
```
        if let Ok(opening_proof) = opening_proof {
            // Code currently in the `else` clause
        } else {
            transcript = transcript_dup;
            transcript.absorb_base(Field::one());
        }
```